src/H5Mpublic.h

```diff
@@ -68,112 +72,116 @@ typedef enum H5VL_map_specific_t {
-        hbool_t     exists;          /* Flag indicating whether key exists in map (OUT) */
+        hid_t       key_mem_type_id; /**< Memory datatype for key */
+        const void *key;             /**< Pointer to key */
+        hbool_t     exists;          /**< Flag indicating whether key exists in map (OUT) */
     } exists;
 
-    /* H5VL_MAP_PUT */
+    /** H5VL_MAP_PUT */
     struct {
-        hid_t       key_mem_type_id;   /* Memory datatype for key */
-        const void *key;               /* Pointer to key */
-        hid_t       value_mem_type_id; /* Memory datatype for value */
-        const void *value;             /* Pointer to value */
+        hid_t       key_mem_type_id;   /**< Memory datatype for key */
+        const void *key;               /**< Pointer to key */
+        hid_t       value_mem_type_id; /**< Memory datatype for value */
+        const void *value;             /**< Pointer to value */
     } put;
 
-    /* H5VL_MAP_GET */
+    /** H5VL_MAP_GET */
     struct {
-        H5VL_map_get_t get_type; /* 'get' operation to perform */
+        H5VL_map_get_t get_type; /**< 'get' operation to perform */
 
-        /* Parameters for each operation */
+        /** Parameters for each operation */
         union {
-            /* H5VL_MAP_GET_MAPL */
+            /** H5VL_MAP_GET_MAPL */
             struct {
-                hid_t mapl_id; /* Map access property list ID (OUT) */
+                hid_t mapl_id; /**< Get map access property list ID (OUT) */
             } get_mapl;
 
-            /* H5VL_MAP_GET_MCPL */
+            /** H5VL_MAP_GET_MCPL */
             struct {
-                hid_t mcpl_id; /* Map creation property list ID (OUT) */
+                hid_t mcpl_id; /**< Get map creation property list ID (OUT) */
             } get_mcpl;
 
-            /* H5VL_MAP_GET_KEY_TYPE */
+            /** H5VL_MAP_GET_KEY_TYPE */
             struct {
-                hid_t type_id; /* Datatype ID for map's keys (OUT) */
+                hid_t type_id; /**< Get datatype ID for map's keys (OUT) */
             } get_key_type;
 
-            /* H5VL_MAP_GET_VAL_TYPE */
+            /** H5VL_MAP_GET_VAL_TYPE */
             struct {
-                hid_t type_id; /* Datatype ID for map's values (OUT) */
+                hid_t type_id; /**< Get datatype ID for map's values (OUT) */
             } get_val_type;
 
-            /* H5VL_MAP_GET_COUNT */
+            /** H5VL_MAP_GET_COUNT */
             struct {
-                hsize_t count; /* # of KV pairs in map (OUT) */
+                hsize_t count; /**< Get number of key-value pairs in the map (OUT) */
             } get_count;
         } args;
     } get;
 
-    /* H5VL_MAP_SPECIFIC */
+    /** H5VL_MAP_SPECIFIC */
     struct {
-        H5VL_map_specific_t specific_type; /* 'specific' operation to perform */
+        H5VL_map_specific_t specific_type;
+        /**< 'specific' operation to perform, which are 'iterate' and 'del' currently */
```

Comment:
I wouldn't mention specific operations here, so that we don't forget to update this in the future.